### PR TITLE
Do not create default SymbolicValue for operations

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs [hunk 1]
```diff
@@ -131,9 +127,6 @@ private void NotifyExecutionCompleted()
             }
         }
 
-        private SymbolicValue CreateSymbolicValue() =>
```
Comment (pavel-mikula-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs:134]: And it's gone! 😄

**Comment by pavel-mikula-sonarsource** [comment]
Converting to draft so S2222 is merged and `SetOperationConstraint` can be used there.

File: analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs [hunk 2]
```diff
@@ -81,21 +81,17 @@ private IEnumerable<ExplodedNode> ProcessBranching(ExplodedNode node)
 
         private IEnumerable<ExplodedNode> ProcessOperation(ExplodedNode node)
         {
-            var context = new SymbolicContext(symbolicValueCounter, node.Operation, node.State.SetOperationValue(node.Operation, CreateSymbolicValue()));
+            var context = new SymbolicContext(symbolicValueCounter, node.Operation, node.State);
             context = InvokeChecks(context, x => x.PreProcess);
-            if (context == null)
+            if (context != null)
```
Comment (costin-zaharia-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs:86]: I'm not always a fan but you sometimes prefer to have this:
```
if (InvokeChecks(new SymbolicContext(symbolicValueCounter, node.Operation, node.State), x => x.PreProcess) is {} context)
```

File: analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/ProgramStateTest.cs [hunk 3]
```diff
@@ -141,6 +142,25 @@ public void SetOperationValue_Overrides()
             sut[operation].Should().Be(value2);
         }
 
+        [TestMethod]
+        public void SetOperationValue_Null_ReturnsNull()
```
Comment (costin-zaharia-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/ProgramStateTest.cs:146]: I would like to be explicit here:
```suggestion
        public void SetOperationValue_NullValue_ReturnsNull()
```

File: analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/ProgramStateTest.cs [hunk 4]
```diff
@@ -141,6 +142,25 @@ public void SetOperationValue_Overrides()
...
+        {
+            var operation = TestHelper.CompileCfgBodyCS("var x = 42;").Blocks[1].Operations[0];
+            var sut = ProgramState.Empty;
+
+            sut[operation].Should().BeNull();
+            sut = sut.SetOperationValue(operation, null);
+            sut[operation].Should().Be(null);
```
Comment (costin-zaharia-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/ProgramStateTest.cs:153]: would be nice to be consistent:
```suggestion
            sut[operation].Should().BeNull();
```

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM! I've added though 2 small comments

**Pull request merged**